apps/hash-graph/bin/cli/src/subcommand.rs

```diff
@@ -33,3 +36,30 @@ pub enum Subcommand {
     #[cfg(all(hash_graph_test_environment, feature = "test-server"))]
     TestServer(TestServerArgs),
 }
+
+impl Subcommand {
+    async fn delegate(self) -> Result<(), GraphError> {
+        match self {
+            Self::Server(args) => server(args).await,
+            Self::Migrate(args) => migrate(args).await,
+            Self::TypeFetcher(args) => type_fetcher(args).await,
+            Self::Completions(ref args) => {
+                completions(args);
+                Ok(())
+            }
+            Self::Snapshot(args) => snapshot(args).await,
+            #[cfg(all(hash_graph_test_environment, feature = "test-server"))]
+            Self::TestServer(args) => test_server(args).await,
+        }
+    }
+
+    pub(crate) fn execute(self) -> Result<(), GraphError> {
+        // create a runtime
+        let runtime = tokio::runtime::Builder::new_multi_thread()
+            .enable_all()
+            .build()
+            .expect("failed to create runtime");
+
+        runtime.block_on(self.delegate())
```

Comment:
Why do we block on `delegate` if not every subcommand in `delegate` is asynchronous? The `completions` subcommand does not require a runtime at all. I know that we previously had a similar behavior with `#[tokio::main]` but as we don't have this anymore we also don't need to spin up a runtime for that.

Reply:
made optional in [8e62433](https://github.com/hashintel/hash/pull/3091/commits/8e624331872d7bb6331f7e7bfd70a877aa182e66)